manifest.go

```diff
@@ -117,6 +121,9 @@ func validateManifest(s string) ([]error, error) {
 								warns = append(warns, fmt.Errorf("Invalid key %q in %q", key, prop))
 							}
 						}
+						if !ruleProvided && len(props) > 0 {
+							warns = append(warns, fmt.Errorf("version rule or source should be provided in %q", prop))
+						}
```

Comment:
I think this should be a dynamic error with `name` in it. So, maybe something like:
```
fmt.Errorf("branch, version, revision, or source should be provided for %q %q", prop, props["name"])
```
Which would look something like:
```
branch, version, revision, or source should be provided for "constraint" "github.com/foo/bar"
```
Since this warning would only show up after checking for missing constraint/override, we are sure that `props["name"]` won't be empty.

Reply:
⚠️ Also, no rule check should only be performed for `constraint` and not `override` because just the `name` attribute is useful in case of overrides. Not the same case with constraints.